Tests/Get-ProGetAsset.Tests.ps1

```diff
@@ -47,72 +47,59 @@ function ThenListShouldBeReturned
     foreach($file in $Name)
     {
         it ('should return a file name that matches ''{0}''' -f $file){
-            $asset | Where-Object {$_.name -match $file } 
+            $asset | Where-Object {$_.name -match $file } | Should -not -BeNullOrEmpty
         }
     }
 
 }
 
-function ThenContentShouldBeReturned
+function ThenListShouldBeEmpty
 {
-    param(
-        [string]
-        $WithContent
-    )
-    it ('should return content that matches ''{0}''' -f $WithContent){
-        $asset | Should -Match $WithContent 
+    it 'should return a list that is empty' {
+        $asset | Should -BeNullOrEmpty
     }
 }
-
 function ThenNoErrorShouldBeThrown
 {
     It 'should not throw an error' {
         $Global:Error | Should -BeNullOrEmpty
     }
 }
 
-function ThenErrorShouldBeThrown
-{
-    param(
-        [string]
-        $ExpectedError
-    )
-    It ('should write an error that matches ''{0}''' -f $ExpectedError) {
-        $Global:Error | Where-Object { $_ -match $ExpectedError } | Should -not -BeNullOrEmpty
-    }
-}
-
-function cleanup
-{
-    $script:asset = Get-ProGetAsset -Session $session -AssetDirectory $progetAssetDirectory
-    foreach($file in $asset)
-    {
-        Remove-ProGetAsset -Session $session -AssetDirectory $progetAssetDirectory -AssetName $file.name
-    }
-    $script:asset = $null
-}
 Describe 'Get-ProGetAsset.when list of assets is returned'{

```

Comment:
Please add some tests around assets in sub-directories, e.g. `dir/dir2/file`.

Reply:
Done, thank you for catching this my original implementation did not work correctly with subdirectories. this is fixed now with tests added for Get-ProgetAsset, Remove-ProGetAsset and Set-ProGetAsset